File: Loenn/entities/darkMirror.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local darkMirror = {}

darkMirror.name = "MaggyHelper/DarkMirror"
darkMirror.depth = -50
darkMirror.fieldInformation = {
    health = {
        minimumValue = 1
    }
}
darkMirror.placements = {
    {
        name = "normal",
        data = {
            health = 2,
            revealsSecret = false
        }
    },
    {
        name = "secret",
        data = {
            health = 2,
            revealsSecret = true
        }
    }
}

function darkMirror.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/dark_mirror/intact", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.0, 0.0, 0.4, 0.8}
    return sprite
end

function darkMirror.selection(room, entity)
    return utils.rectangle(entity.x - 24, entity.y - 64, 48, 64)
end

return darkMirror

File: framework/extracted/Ahorn/entities/DarkMirror.jl
module DarkMirror

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/DarkMirror" DarkMirror(x::Integer, y::Integer, health::Integer=2, revealsSecret::Bool=false)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(DarkMirror),
    "secret" => Ahorn.EntityPlacement(DarkMirror)
)

function Ahorn.selection(entity::DarkMirror)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::DarkMirror, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
